feat: Add note to tree shaking docs (#8143)
getsentry/sentry-docs: src/platforms/javascript/common/configuration/tree-shaking/index.mdx

> The Sentry SDK ships with code that is not strictly required for it to collect your errors. This includes, for example, code to debug your Sentry configuration or code to enable performance monitoring. While debug code can be very useful in development environments, it's not typically necessary to include it in your production bundles where it takes up valuable space. The JavaScript SDK includes a special flags in its CommonJS and ESM distributions, which can be used to facilitate tree shaking (removal) of such code during the build process.

Reviewer: ```suggestion
The Sentry SDK ships with code that is not strictly required for it to collect your errors. This includes code to debug your Sentry configuration or code to enable performance monitoring, for example. While debug code can be very useful in development environments, it's not typically necessary to include it in your production bundles where it takes up valuable space. The JavaScript SDK includes special flags in its CommonJS and ESM distributions, which can be used to facilitate tree shaking (removal) of this kind of code during the build process.
```

<Note>

>   Anything that you do not import & use will <i>automatically</i> be tree shaken
>   away. This means that optional integrations like Replay, BrowserTracing,

Author: ```suggestion
  Anything that you do not import & use will <i>automatically</i> be tree shaken
  away when using any modern bundler like Webpack, Rollup, Vite or similar. This means that optional integrations like Replay, BrowserTracing,
```

@lforst WDYT about this?

>   BrowserProfiling or any unusued utility methods will not be included in your
>   bundle unless you import and use them yourself. The rest of this page relates
>   to ways to tree shake internal SDK code, which is not strictly required in
>   case you do not use certain features.

Reviewer: ```suggestion
  BrowserProfiling, and any unused utility methods won't be included in your
  bundle unless you import and use them yourself. The rest of this page relates
  to ways to tree shake internal SDK code, which isn't strictly required unless
  you use certain features.
```
